cachelib/allocator/CacheAllocator-inl.h

```diff
@@ -2468,7 +2340,7 @@ void CacheAllocator<CacheTrait>::releaseSlabImpl(
     // Need to mark an item for release before proceeding
     // If we can't mark as moving, it means the item is already freed
     const bool isAlreadyFreed =
-        !markMovingForSlabRelease(releaseContext, alloc, throttler);
+        !markExclusiveForSlabRelease(releaseContext, alloc, throttler);
```

Comment:
This is no longer safe. The original intent is that only a freed item cannot be marked as moving and no allocations can allocate a freed item once that slab was set to be released.
With this change, we can get false here if the item was already marked as exclusive (e.g. from an eviction thread). This means here we would treat that item as one already freed, which isn't the case.

Reply:
I think this cannot happen. If we fail to mark the item as exclusive inside `processAllocForRelease` we will just spin on the `while(true)` inside `markExclusiveForSlabRelease`

Reply:
^ yup you're right.